src/Device/SmartDeviceSensors.cpp

```
@@ -99,7 +127,23 @@ DeviceDescriptor::OnLocationSensor(std::chrono::system_clock::time_point time,
+  const bool log_nmea =
+    nmea_logger != nullptr &&
+    nmea_logger->IsEnabled() &&
+    n_satellites >= MIN_SATELLITES_FOR_3D_FIX;
+  NMEAInfo log_basic{};
+  if (log_nmea)
+    log_basic = basic;
+
   e.Commit();
+
+  if (log_nmea) {
+    char body[128];
```

Cppcheck (2.21.0)

[error] Dereference of an invalid iterator

(derefInvalidIterator)